### PR TITLE
refactor: project to use gulpfile for js minification; fix: working clan leaving; fix: profile mode selection

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

File: web/package.json [hunk 1]
```diff
@@ -22,5 +22,8 @@
     "jquery": "^3.1.1",
     "timeago": "^1.5.3",
     "vue": "^2.1.4"
+  },
+  "overrides": {
+    "graceful-fs": "^4.2.10"
```
Comment (cmyui) [inline web/package.json:27]: https://stackoverflow.com/a/58394828

File: web/static/js/pages/profile.js [hunk 2]
```diff
@@ -860,4 +842,31 @@ function beatmapLink(type, id) {
...
+	if (mode === 3) {
+		$(".simple-banner-swtich[data-rx='1']").addClass('disabled')
+		$(".simple-banner-swtich[data-rx='2']").addClass('disabled')
+	} else if (mode === 1 || mode === 2) {
+		$(".simple-banner-swtich[data-rx='2']").addClass('disabled')
+	} else {
+		$(".simple-banner-swtich[data-rx='1']").removeClass('disabled')
+		$(".simple-banner-swtich[data-rx='2']").removeClass('disabled')
+	}
 }
```
Comment (cmyui) [inline web/static/js/pages/profile.js]: @lenforiee hey, why did this get reverted?

pretty sure we were using `toggleModeAvailability`, right? or am i misremembering

Comment (tsunyoku) [reply web/static/js/pages/profile.js]: `swtich` what the hell

Comment (cmyui) [reply web/static/js/pages/profile.js]: ah yes found the issue - toggleModeAvailability was defined in dist.min.js but not in the other pages (like clan.js)

**Pull request merged**